File: Difficulty: Medium/Container With Most Water/container-with-most-water.py
class Solution:
    def maxWater(self, arr):
        # code here
        l,r=0,len(arr)-1
        res=0
        while(l<r):
            area =(r-l)*min(arr[l],arr[r])
            res = max(area,res)
            if(arr[l]>arr[r]) : r-=1
            elif(arr[l]<=arr[r]): l+=1
        return res

File: Difficulty: Medium/Container With Most Water/container-with-most-water.java
class Solution {
    public int maxWater(int arr[]) {
        // Code Here
        int l=0,r=arr.length-1;
        int res = 0;
        while(l<r){
            int area = (r-l)*Math.min(arr[r],arr[l]);
            res = Math.max(area,res);
            
            if(arr[r]<arr[l]) r--;
            else if(arr[l]<arr[r]) l++;
            else r--;
        }return res;
    }
}